### PR TITLE
Updating linux tables to NewExecAndParse

### PR DESCRIPTION
**Pull request by cesarfda**
This pull request introduces parsers for JSON and key-value data formats, along with comprehensive test cases and sample data files. The implementation includes singleton instances, parsing logic, and handling of edge cases such as malformed data and duplicate keys.

### Migrated Tables:

- kolide_nmcli_wifi
- kolide_lsblk
- kolide_wsone_uem_status_enroll
- kolide_wsone_uem_status_dependency
- kolide_wsone_uem_status_profile
- kolide_zypper_upgradeable_packages
- kolide_zypper_upgradeable_patches
- kolide_nftables

### JSON Parser Implementation:

* **Parser creation and singleton instance**: Added `New` function to initialize a JSON parser and a singleton instance `Parser`. (`ee/tables/execparsers/json/json.go`, [ee/tables/execparsers/json/json.goR1-R21](diffhunk://#diff-7bd8e29b629d3ecce2e3332d71a667e3d9d476210dfd5ead29b9abd6a6756c66R1-R21))
* **Parsing logic**: Implemented `parseJson` method to parse JSON data, handle UTF-16 to UTF-8 conversion, and validate JSON structure. (`ee/tables/execparsers/json/parser.go`, [ee/tables/execparsers/json/parser.goR1-R47](diffhunk://#diff-253108c7d490d88d454b6d68c6fb4394bf1398da40e40bcfb3fb87f4f8ec1788R1-R47))
* Example result from interactive:
* image

### JSON Parser Tests:

* **Test suite**: Added unit tests covering various JSON parsing scenarios, including valid data, malformed JSON, and edge cases. (`ee/tables/execparsers/json/parse_test.go`, [ee/tables/execparsers/json/parse_test.goR1-R243](diffhunk://#diff-c188cbc4bbb1bfc83375ea376a2c843d09d42b85b6b4ab854d2af0c54f2c99e6R1-R243))
* **Sample data files**: Included JSON sample files for `lsblk` and `nftables` data to support test cases. (`ee/tables/execparsers/json/test-data/kolide_lsblk.json`, [[1]](diffhunk://#diff-a831fcfb9f11e44e34ff65579311fa0f4bfd5baafe92d7a46732161717f06331R1-R240); `ee/tables/execparsers/json/test-data/kolide_nftables.json`, [[2]](diffhunk://#diff-a11d7fd837f4790746e7fa9ebe52ac07ee37e304c5b0618619aaca2fa15f2ebcR1)

### Key-Value Parser Implementation:

* **Parser creation and singleton instance**: Added `New` and `NewWithDelimiter` functions for initializing key-value parsers with default or custom delimiters, and a singleton instance `Parser`. (`ee/tables/execparsers/key_value/key_value.go`, [ee/tables/execparsers/key_value/key_value.goR1-R41](diffhunk://#diff-1fca85de6f28f0ad8769645262277fc8a8eaa62e88f57db18a3073f805ae5ec7R1-R41))
* **Parsing logic**: Implemented `parseKeyValue` method to parse key-value pairs, handle nested keys, and manage duplicate keys by converting them into arrays. (`ee/tables/execparsers/key_value/parser.go`, [ee/tables/execparsers/key_value/parser.goR1-R115](diffhunk://#diff-26d6b19eb5bd2b5fd630111f6795eb1dee22d9c73908093367ca0e66df89732aR1-R115))

### CODE REVIEW

File: ee/tables/execparsers/json/json.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+//go:build darwin || linux
+// +build darwin linux
```
Comment (RebeccaMahany) [inline ee/tables/execparsers/json/json.go:2]: Why is this darwin/linux only? (Usually, unless we aren't able to, we keep files cross-platform)

File: ee/tables/execparsers/json/parser.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		if err := json.Unmarshal(rawdata, &data); err != nil {
+			return nil, fmt.Errorf("unmarshalling json: %w", err)
+		}
+		return data, nil
+	}
+
+	// We don't have valid json data, so try to convert possible utf16 data to utf8
+	rawdata, _, err = transform.Bytes(unicode.UTF16(unicode.LittleEndian, unicode.UseBOM).NewDecoder(), rawdata)
+	if err != nil {
+		return nil, errors.New("invalid json. (Despite attempted transform from utf16 to utf8)")
```
Comment (RebeccaMahany) [inline ee/tables/execparsers/json/parser.go:38]: I know that this is how it is in dataflatten too, but I think a) the message is a little misleading and b) we should include the err as well -- 

```golang
return nil, fmt.Errorf("transforming invalid json from utf16 to utf8: %w", err)
```

File: ee/tables/execparsers/json/parser.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	}
+
+	// Try to unmarshal the transformed data
+	if err := json.Unmarshal(rawdata, &data); err != nil {
+		return nil, fmt.Errorf("unmarshalling json: %w", err)
```
Comment (RebeccaMahany) [inline ee/tables/execparsers/json/parser.go:43]: Nitpick -- maybe worth adding a little bit of extra context to disambiguate between this error and the one on line 30 above --

```suggestion
		return nil, fmt.Errorf("unmarshalling json after transforming from utf16 to utf8: %w", err)
```

File: ee/tables/execparsers/key_value/key_value.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
+//go:build darwin || linux
+// +build darwin linux
```
Comment (RebeccaMahany) [inline ee/tables/execparsers/key_value/key_value.go:2]: Same q here, why darwin/linux only? Seems like it could be useful/applicable for Windows too

File: ee/tables/execparsers/key_value/test-data/nmcli.txt [hunk 5]
```diff
@@ -0,0 +1,663 @@
+﻿NAME:                                                                      AP[1]
```
Comment (RebeccaMahany) [inline ee/tables/execparsers/key_value/test-data/nmcli.txt:1]: I think we should remove this test data and replace with something anonymized

File: ee/tables/execparsers/key_value/key_value.go [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+	"io"
+)
+
+// parser implements the parser interface for key-value data
+type parser struct {
+	kvDelimiter string
+}
+
+// Parser is the singleton instance of the parser
+var Parser = New()
```
Comment (RebeccaMahany) [inline ee/tables/execparsers/key_value/key_value.go:16]: Do we want a singleton instance to be available here? Wondering if we might accidentally use this in the future in a couple places and not realize when calling `SetDelimiter` that we've updated the delimiter for both tables

File: ee/tables/execparsers/mapxml/mapxml.go [hunk 7]
```diff
@@ -1,5 +1,5 @@
-//go:build darwin
-// +build darwin
+//go:build darwin || linux
```
Comment (directionless) [inline ee/tables/execparsers/mapxml/mapxml.go:1]: Probably just build everywhere

File: ee/tables/execparsers/key_value/test-data/nmcli.txt (hunk 5 above)
Comment (directionless) [reply ee/tables/execparsers/key_value/test-data/nmcli.txt:1]: Scamble the BSSID and SSID values

**Review by RebeccaMahany — APPROVED** [review]
